docs/tools/testing/unit_testing.md

```diff
@@ -1,27 +1,275 @@
 
-Unit tests are located in Mbed OS repository under `UNITTESTS`. Each unit test uses an identical directory tree structure to the file to be tested. This makes it easier to find unit tests for a particular class or a module. For example, if the file to be tested is `rtos/Semaphore.cpp`, then all the test files are in the `UNITTESTS/rtos/Semaphore` directory.
+Unit tests are located in Mbed OS repository under `UNITTESTS` folder. Unit test files should use identical directory path to the file under test. This makes it easier to find unit tests for a particular class or a module. For example, if the file under test is `rtos/Semaphore.cpp`, then all the test files should be in the `UNITTESTS/rtos/Semaphore` directory. Each test suite needs to have their own `unittest.cmake` file for test configuration.
 
-##### Test discovery
+#### Test discovery
 
 Registering unit tests for running is automatic, and the test runner handles registration. However, test files are not automatically assigned to be built. We build unit tests by using a separate build system, which searches for unit tests under the `UNITTESTS` directory.
 
 For the build system to find and build any test suite automatically, a unit test configuration file named `unittest.cmake` is required to be included with each unit test. This configuration file contains a name for the test and other source files the build requires.
 
-##### Test names
+#### Test names
 
 Each test suite requires a name to be configured in the `unittest.cmake` file. This name is used for generated files and when running a subset of tests.
 
-#### Building, running and writing tests
+### Writing unit tests
+
+Each unit test suite requires at least two files to be created:
+1. C++ unit test source file (e.g. `test_NetworkInterface.cpp`)
+2. CMake module file for unit test definition (`unittest.cmake`)
+
+A unit test definition file `unittest.cmake` requires variables to be set for a test to be configured. File source paths in `unittest.cmake` files need to be relative to the unit test folder and `CMakeLists.txt`.
+
+* **TEST_SUITE_NAME** - Unique identifier for the test suite. Use naming convention *PATH-TO-THE-TESTABLE-FILE* e.g. *features-netsocket-InternetSocket*
+* **unittest-includes** - Include paths for headers needed to build the tests in addition to the base include paths listed in CMakeLists.txt. Optional.
+* **unittest-sources** - Mbed OS source files and stubs included for the build.
+* **unittest-test-sources** - Unit test source files.
+
+##### Creating unit tests files with Mbed CLI
+
+To learn how to generate test files, please see [a handbook section about using unit tests with Mbed CLI](https://os.mbed.com/docs/latest/tools/test-and-debug.html#unit-testing). The generator can only create the files required for a unit test. It does not write unit tests automatically nor does it handle source dependencies.
+
+##### Create files manually
+
+For example to create a unit test for `rtos/Semaphore.cpp`:
```

Comment:
Don't start with "for example" if there was nothing but a title before it. It's totally jarring. Say something like "Here is an example of".